kernel/src/lib.rs

```diff
@@ -340,6 +344,32 @@ pub trait ExpressionHandler: AsAny {
         expression: Expression,
         output_type: DataType,
     ) -> Arc<dyn ExpressionEvaluator>;
+
+    /// Create a single-row null-value [`EngineData`] of a single column of type `output_type`.
+    // NOTE: we should probably allow DataType instead of SchemaRef, but can expand that in the
+    // future.
+    fn null_row(&self, output_type: SchemaRef) -> DeltaResult<Box<dyn EngineData>>;
```

Comment:
nit:
```suggestion
    fn null_row(&self, output_schema: SchemaRef) -> DeltaResult<Box<dyn EngineData>>;
```

We generally use `type` for `DataType`